TestSRTPDrainLeak occasionally fails

<!--
  Come join our community on Slack https://pion.ly/slack for quicker support and more resources

  We also provide a Knowledge Base https://pion.ly/knowledge-base that covers Pion and general WebRTC
  
  We maintain a FAQ of common issues and questions https://pion.ly/knowledge-base/pion-basics/faq/

 Please answer these questions before submitting your issue. Thanks!
-->

### Your environment.
  - Version: 0e257f6
  - Browser: n/a
  - Other Information

Travis CI [build #3831](https://travis-ci.org/pion/webrtc/builds/582373663?utm_source=github_status&utm_medium=notification)
```
=== RUN   TestSRTPDrainLeak
ice ERROR: 2019/09/08 17:27:17 error processing checkCandidatesTimeout handler the agent is closed
goroutine profile: total 36
10 @ 0x8076614 0x80713c4 0x807089b 0x80bf827 0x80c0fbd 0x80c0f9e 0x81c8ef8 0x823c53c 0x823fb42 0x823fb21 0x8243801 0x809f0c1
#	0x807089a	internal/poll.runtime_pollWait+0x4a			/usr/local/go/src/runtime/netpoll.go:184
#	0x80bf826	internal/poll.(*pollDesc).wait+0x36			/usr/local/go/src/internal/poll/fd_poll_runtime.go:87
#	0x80c0fbc	internal/poll.(*pollDesc).waitRead+0xec			/usr/local/go/src/internal/poll/fd_poll_runtime.go:92
#	0x80c0f9d	internal/poll.(*FD).RawRead+0xcd			/usr/local/go/src/internal/poll/fd_unix.go:534
#	0x81c8ef7	net.(*rawConn).Read+0x47				/usr/local/go/src/net/rawconn.go:43
#	0x823c53b	golang.org/x/net/internal/socket.(*Conn).recvMsg+0x18b	/go/pkg/mod/golang.org/x/net@v0.0.0-20190628185345-da137c7871d7/internal/socket/rawconn_msg.go:31
#	0x823fb41	golang.org/x/net/internal/socket.(*Conn).RecvMsg+0x141	/go/pkg/mod/golang.org/x/net@v0.0.0-20190628185345-da137c7871d7/internal/socket/socket.go:255
#	0x823fb20	golang.org/x/net/ipv4.(*payloadHandler).ReadFrom+0x120	/go/pkg/mod/golang.org/x/net@v0.0.0-20190628185345-da137c7871d7/ipv4/payload_cmsg.go:31
#	0x8243800	github.com/pion/mdns.(*Conn).start+0xf0			/go/pkg/mod/github.com/pion/mdns@v0.0.3/conn.go:254
9 @ 0x8076614 0x80713c4 0x807089b 0x80bf827 0x80c03b6 0x80c039b 0x81ba79f 0x81ce0ef 0x81cceb7 0x82632d0 0x809f0c1
#	0x807089a	internal/poll.runtime_pollWait+0x4a			/usr/local/go/src/runtime/netpoll.go:184
#	0x80bf826	internal/poll.(*pollDesc).wait+0x36			/usr/local/go/src/internal/poll/fd_poll_runtime.go:87
#	0x80c03b5	internal/poll.(*pollDesc).waitRead+0x165		/usr/local/go/src/internal/poll/fd_poll_runtime.go:92
#	0x80c039a	internal/poll.(*FD).ReadFrom+0x14a			/usr/local/go/src/internal/poll/fd_unix.go:219
#	0x81ba79e	net.(*netFD).readFrom+0x3e				/usr/local/go/src/net/fd_unix.go:208
#	0x81ce0ee	net.(*UDPConn).readFrom+0x3e				/usr/local/go/src/net/udpsock_posix.go:47
#	0x81cceb6	net.(*UDPConn).ReadFrom+0x46				/usr/local/go/src/net/udpsock.go:121
#	0x82632cf	github.com/pion/ice.(*candidateBase).recvLoop+0xef	/go/pkg/mod/github.com/pion/ice@v0.5.14/candidate_base.go:92
9 @ 0x8076614 0x8084b99 0x825fbba 0x809f0c1
#	0x825fbb9	github.com/pion/ice.(*Agent).taskLoop+0x1b9	/go/pkg/mod/github.com/pion/ice@v0.5.14/agent.go:635
1 @ 0x8076614 0x804d325 0x804d300 0x804d0f4 0x82a855a 0x82ab15b 0x809f0c1
#	0x82a8559	github.com/pion/srtp.(*SessionSRTCP).decrypt+0x209	/go/pkg/mod/github.com/pion/srtp@v1.2.6/session_srtcp.go:143
#	0x82ab15a	github.com/pion/srtp.(*session).start.func1+0xfa	/go/pkg/mod/github.com/pion/srtp@v1.2.6/session.go:132
1 @ 0x8076614 0x804d325 0x804d300 0x804d0f4 0x842464a 0x809f0c1
#	0x8424649	github.com/pion/webrtc/v2.TestPeerConnection_EventHandlers_Go.func5+0x29	/go/src/github.com/pion/webrtc/peerconnection_go_test.go:272
1 @ 0x8076614 0x804dd82 0x804dd5d 0x804dacc 0x811132b 0x811417a 0x8110ffa 0x811239d 0x811164b 0x8435db4 0x8076242 0x809f0c1
#	0x811132a	testing.(*T).Run+0x2ca		/usr/local/go/src/testing/testing.go:961
#	0x8114179	testing.runTests.func1+0x59	/usr/local/go/src/testing/testing.go:1202
#	0x8110ff9	testing.tRunner+0x99		/usr/local/go/src/testing/testing.go:909
#	0x811239c	testing.runTests+0x22c		/usr/local/go/src/testing/testing.go:1200
#	0x811164a	testing.(*M).Run+0x13a		/usr/local/go/src/testing/testing.go:1117
#	0x8435db3	main.main+0x163			_testmain.go:534
#	0x8076241	runtime.main+0x201		/usr/local/go/src/runtime/proc.go:203
1 @ 0x8076614 0x804dd82 0x804dd5d 0x804dacc 0x82a77ba 0x83e269f 0x83e2693 0x83f1797 0x84074bc 0x8110ffa 0x809f0c1
#	0x82a77b9	github.com/pion/srtp.(*session).close+0x59		/go/pkg/mod/github.com/pion/srtp@v1.2.6/session.go:95
#	0x83e269e	github.com/pion/srtp.(*SessionSRTCP).Close+0x1fe	/go/pkg/mod/github.com/pion/srtp@v1.2.6/session_srtcp.go:89
#	0x83e2692	github.com/pion/webrtc/v2.(*DTLSTransport).Stop+0x1f2	/go/src/github.com/pion/webrtc/dtlstransport.go:298
#	0x83f1796	github.com/pion/webrtc/v2.(*PeerConnection).Close+0x86	/go/src/github.com/pion/webrtc/peerconnection.go:1600
#	0x84074bb	github.com/pion/webrtc/v2.TestSRTPDrainLeak+0x78b	/go/src/github.com/pion/webrtc/peerconnection_media_test.go:129
#	0x8110ff9	testing.tRunner+0x99					/usr/local/go/src/testing/testing.go:909
1 @ 0x8076614 0x8084b99 0x825fadb 0x809f0c1
#	0x825fada	github.com/pion/ice.(*Agent).taskLoop+0xda	/go/pkg/mod/github.com/pion/ice@v0.5.14/agent.go:622
1 @ 0x8076614 0x8084b99 0x826a81a 0x83e69c5 0x83e6973 0x841ae58 0x809f0c1
#	0x826a819	github.com/pion/ice.(*Agent).connect+0x129					/go/pkg/mod/github.com/pion/ice@v0.5.14/transport.go:57
#	0x83e69c4	github.com/pion/ice.(*Agent).Accept+0x234					/go/pkg/mod/github.com/pion/ice@v0.5.14/transport.go:22
#	0x83e6972	github.com/pion/webrtc/v2.(*ICETransport).Start+0x1e2				/go/src/github.com/pion/webrtc/icetransport.go:124
#	0x841ae57	github.com/pion/webrtc/v2.(*PeerConnection).SetRemoteDescription.func3+0xc7	/go/src/github.com/pion/webrtc/peerconnection.go:946
1 @ 0x8076614 0x80856ad 0x808569e 0x80853fa 0x80a92b6 0x83e1d21 0x83efcb7 0x809f0c1
#	0x80853f9	sync.runtime_SemacquireMutex+0x39				/usr/local/go/src/runtime/sema.go:71
#	0x80a92b5	sync.(*RWMutex).RLock+0x55					/usr/local/go/src/sync/rwmutex.go:50
#	0x83e1d20	github.com/pion/webrtc/v2.(*DTLSTransport).getSRTCPSession+0x30	/go/src/github.com/pion/webrtc/dtlstransport.go:194
#	0x83efcb6	github.com/pion/webrtc/v2.(*PeerConnection).drainSRTP+0xa6	/go/src/github.com/pion/webrtc/peerconnection.go:1200
1 @ 0x814531c 0x814518a 0x81421f3 0x83dd357 0x809f0c1
#	0x814531b	runtime/pprof.writeRuntimeProfile+0x7b			/usr/local/go/src/runtime/pprof/pprof.go:708
#	0x8145189	runtime/pprof.writeGoroutine+0x79			/usr/local/go/src/runtime/pprof/pprof.go:670
#	0x81421f2	runtime/pprof.(*Profile).WriteTo+0x2c2			/usr/local/go/src/runtime/pprof/pprof.go:329
#	0x83dd356	github.com/pion/transport/test.TimeOut.func1+0x56	/go/pkg/mod/github.com/pion/transport@v0.8.7/test/util.go:18
panic: timeout
goroutine 979 [running]:
github.com/pion/transport/test.TimeOut.func1()
	/go/pkg/mod/github.com/pion/transport@v0.8.7/test/util.go:21 +0xda
created by time.goFunc
	/usr/local/go/src/time/sleep.go:168 +0x37
FAIL	github.com/pion/webrtc/v2	136.934s
```

--- a/peerconnection_media_test.go
+++ b/peerconnection_media_test.go
@@ -4,6 +4,7 @@ package webrtc
 
 import (
 	"bytes"
+	"context"
 	"fmt"
 	"io"
 	"math/rand"
@@ -38,21 +39,22 @@ func TestSRTPDrainLeak(t *testing.T) {
 	report := test.CheckRoutines(t)
 	defer report()
 
-	sawRTCPDrainMessage := make(chan bool, 1)
-	sawRTPDrainmessage := make(chan bool, 1)
+	sawRTPDrainMessage := &atomicBool{}
+	sawRTCPDrainMessage := &atomicBool{}
+	seenBothMessages, closeFunc := context.WithCancel(context.Background())
 
 	s := SettingEngine{
 		LoggerFactory: testCatchAllLoggerFactory{
 			callback: func(msg string) {
 				if strings.Contains(msg, "Incoming unhandled RTP ssrc") {
-					select {
-					case sawRTPDrainmessage <- true:
-					default:
+					sawRTPDrainMessage.set(true)
+					if sawRTPDrainMessage.get() && sawRTCPDrainMessage.get() {
+						closeFunc()
 					}
 				} else if strings.Contains(msg, "Incoming unhandled RTCP ssrc") {
-					select {
-					case sawRTCPDrainMessage <- true:
-					default:
+					sawRTCPDrainMessage.set(true)
+					if sawRTPDrainMessage.get() && sawRTCPDrainMessage.get() {
+						closeFunc()
 					}
 				}
 			},
@@ -99,27 +101,36 @@ func TestSRTPDrainLeak(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// Send 5 RTP/RTCP packets with different SSRCes
-	var i uint32
-	for i = 0; i < 5; i++ {
-		if _, err = srtpStream.WriteRTP(&rtp.Header{Version: 2, SSRC: i}, []byte{0x00, 0x01, 0x03}); err != nil {
-			t.Fatal(err)
-		}
-	}
-	for i = 0; i < 5; i++ {
-		var raw []byte
-		raw, err = rtcp.Marshal([]rtcp.Packet{&rtcp.PictureLossIndication{MediaSSRC: i}})
-		if err != nil {
-			t.Fatal(err)
-		}
+	func() {
+		var rtpSsrc uint32
+		var rtcpSsrc uint32
 
-		if _, err = srtcpStream.Write(raw); err != nil {
-			t.Fatal(err)
-		}
-	}
+		for {
+			select {
+			case <-seenBothMessages.Done():
+				return
+			case <-time.After(200 * time.Millisecond):
+				// Send 5 RTP/RTCP packets with different SSRCes
+				for ; rtpSsrc%5 == 0; rtpSsrc++ {
+					if _, err = srtpStream.WriteRTP(&rtp.Header{Version: 2, SSRC: rtpSsrc}, []byte{0x00, 0x01, 0x03}); err != nil {
+						t.Fatal(err)
+					}
+				}
+				for ; rtcpSsrc%5 == 0; rtcpSsrc++ {
+					var raw []byte
+					raw, err = rtcp.Marshal([]rtcp.Packet{&rtcp.PictureLossIndication{MediaSSRC: rtcpSsrc}})
+					if err != nil {
+						t.Fatal(err)
+					}
 
-	<-sawRTCPDrainMessage
-	<-sawRTPDrainmessage
+					if _, err = srtcpStream.Write(raw); err != nil {
+						t.Fatal(err)
+					}
+				}
+			}
+		}
+	}()
+
 	err = pcOffer.Close()
 	if err != nil {
 		t.Fatal(err)

